Fix/IndicationBook.cs

```diff
@@ -0,0 +1,290 @@
+using System;
+using System.Linq;
+using static Fix.Dictionary;
+
+namespace Fix
+{
+    public class IndicationsEventArgs : EventArgs
+    {
+        public IndicationsEventArgs(Indication indication)
+        {
+            Indication = indication;
+        }
+
+        public Indication Indication { get; }
+    }
+
+    public class IndicationBook
+    {
+        #region Events
+
+        public delegate void IndicationDeligate(object sender, IndicationsEventArgs e);
+
+        public event IndicationDeligate? IndicationInserted;
+        public event IndicationDeligate? IndicationUpdated;
+        public event IndicationDeligate? IndicationDeleted;
+
+        protected void OnIndicationInserted(Indication indication)
+        {
+            IndicationInserted?.Invoke(this, new IndicationsEventArgs(indication));
+        }
+
+        protected void OnIndicationUpdated(Indication indication)
+        {
+            IndicationUpdated?.Invoke(this, new IndicationsEventArgs(indication));
+        }
+
+        protected void OnIndicationDeleted(Indication indication)
+        {
+            IndicationDeleted?.Invoke(this, new IndicationsEventArgs(indication));
+        }
+
+        #endregion
+
+        const string StatusMessageHeader = "This message was not processed by the indication book";
+
+        public IndicationCollection Indications { get; } = new();
+
+        public bool DeleteInactiveIndications { get; set; }
+        public int MaximumIndications { get; set; }
+
+        public MessageCollection Messages { get; } = new MessageCollection();
+
+        [Flags]
+        public enum Retain
+        {
+            None = 0,
+            ActiveGTC = 1 << 1,
+            ActiveGTD = 1 << 2
+        }
+
+        public void Clear(Retain options = Retain.None)
+        {
+            if (options == Retain.None)
+            {
+                Indications.Clear();
+            }
+
+            Messages.Clear();
+        }
+
+        public void Delete(Indication indication)
+        {
+            if (Indications.Remove(KeyForIndication(indication)))
+            {
+                OnIndicationDeleted(indication);
+            }
+        }
+
+        public bool Process(Message message)
+        {
+            if (message == null)

```

Comment:
This check is redundant with nullable references now enabled in this code, message cannot be null here.

Reply:
removed.